### PR TITLE
feat: Add Daum Presence

### PR DESCRIPTION
**Pull request by SlowLife1661**
# Daum Presence
image
image
image
image
image
image
image
image
image
image

### CODE REVIEW

**Comment by SlowLife1661** [comment]
I broke it again LOL

File: websites/D/Daum/presence.ts [hunk 1]
```diff
@@ -0,0 +1,394 @@
...
+      "/person/": {
+        service: "DAUM_MOVIE",
+        setPresenceData(){
+          presenceData.details = "Viewing cast:";
+          presenceData.state = document.querySelector('span.txt_tit')?.textContent;
+
+          presenceData.buttons = [{
+            label: "View Cast",
+            url: document.baseURI
+          }];
```
Comment (SlowLife1661) [inline websites/D/Daum/presence.ts:198]: It's a bit weird using this, perhaps Actor will be better? But there are Actor and Actress so am I supposed to use Actor alone LOL but probably

Comment (EncryptedDev) [reply websites/D/Daum/presence.ts:198]: Nah, "View Cast" makes more sense in my head. "actor" implies a singular actor

File: websites/D/Daum/presence.ts [hunk 2]
```diff
@@ -0,0 +1,394 @@
...
+        }
+      },
+      "/person/": {
+        service: "DAUM_MOVIE",
+        setPresenceData(){
+          presenceData.details = "Viewing cast:";
+          presenceData.state = document.querySelector('span.txt_tit')?.textContent;
+
```
Comment (Bas950) [inline websites/D/Daum/presence.ts:194]: shouldnt it be
"Viewing cast of: {show}"

**Review by Bas950 — CHANGES_REQUESTED** [review]
also no support for Daum Cafe?

**Comment by SlowLife1661** [comment]
Ok

File: websites/D/Daum/presence.ts (hunk 2 above)
Comment (SlowLife1661) [reply websites/D/Daum/presence.ts:194]: It's actually one person, I'm probably the one who mislead here LOL

File: websites/D/Daum/dist/metadata.json [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        "movie.daum.net",
+        "news.daum.net",
+        "sports.daum.net",
+        "entertain.daum.net",
+        "auto.daum.net",
+        "mail.daum.net",
+        "blog.daum.net",
+        "top.cafe.daum.net",
+        "cafe.daum.net"
+    ],
```
Comment (ririxidev) [inline websites/D/Daum/dist/metadata.json:26]: Maybe regex?

File: websites/D/Daum/iframe.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+   }
+
+   if (document.location.hostname === "cafe.daum.net")
+    IFrame.send({
+      cafe: { 
+        name: document.querySelector("#cafe_info_outer > div.cafename")?.textContent.trim(),
+        title: (document.querySelector('strong.tit_item') ?? document.querySelector("h3.title"))?.textContent.trim(),
+        article: document.querySelector('strong.tit_info > b')?.textContent.trim()
+      }
+    });
```
Comment (ririxidev) [inline websites/D/Daum/iframe.ts:18]: this is in iframe that you literally need iframe tho?

File: websites/D/Daum/presence.ts [hunk 5]
```diff
@@ -0,0 +1,447 @@
...
+        setPresenceData(){
+          const timestamps = presence.getTimestamps(video?.currentTime, video?.duration);
+          
+          presenceData.details = "Watching video:";
+          presenceData.state = document.querySelector("strong.tit_vod")?.textContent;
+
+          presenceData.smallImageKey = video?.paused ? "pause" : "play";
+          presenceData.smallImageText = video?.paused ? "Paused" : "Playing";
+
+          presenceData.startTimestamp = timestamps[0];
```
Comment (ririxidev) [inline websites/D/Daum/presence.ts:171]: You don't need to specify startTimestamp when you have endTimestamp...

File: websites/D/Daum/presence.ts [hunk 6]
```diff
@@ -0,0 +1,447 @@
...
```
Comment (ririxidev) [inline websites/D/Daum/presence.ts:259]: You don't need to specify startTimestamp when you have endTimestamp...

File: websites/D/Daum/presence.ts [hunk 7]
```diff
@@ -0,0 +1,447 @@
...
+        setPresenceData(){
+          const timestamps = presence.getTimestamps(video?.currentTime, video?.duration);
+          
+          presenceData.details = video?.title ?? document.querySelector('span.inner_tit2')?.textContent;
+          presenceData.state = document.querySelector('a.link_txt')?.textContent.trim();
+
+          presenceData.smallImageKey = video?.paused ? "pause" : "play";
+          presenceData.smallImageText = video?.paused ? "Paused" : "Playing";
+
+          presenceData.startTimestamp = timestamps[0];
```
Comment (ririxidev) [inline websites/D/Daum/presence.ts:284]: You don't need to specify startTimestamp when you have endTimestamp...

File: websites/D/Daum/presence.ts [hunk 8]
```diff
@@ -0,0 +1,447 @@
...
+          | "DAUM_NEWS"
+          | "DAUM_SPORTS"
+          | "DAUM_BLOG"
+          | "DAUM_MAIL"
+          | "DAUM_ENTERTAIN"
+          | "DAUM_CAFE"
+          | "ANY";
+          href?: boolean,
+          hash?: boolean,
+          artitle?: boolean
```
Comment (ririxidev) [inline websites/D/Daum/presence.ts:62]: artitle? Isn't it typo?

File: websites/D/Daum/presence.ts [hunk 9]
```diff
@@ -0,0 +1,447 @@
...
+          if (video){
+            const timestamps = presence.getTimestamps(video.currentTime, video.duration);
+
+            presenceData.details = "Watching video:";
+            presenceData.state = document.querySelector("strong.tit_player")?.textContent;
+
+            presenceData.smallImageKey = video.paused ? "pause" : "play";
+            presenceData.smallImageText = video.paused ? "Paused" : "Playing";
+
+            presenceData.startTimestamp = timestamps[0];
```
Comment (ririxidev) [inline websites/D/Daum/presence.ts:233]: You don't need to specify startTimestamp when you have endTimestamp...

File: websites/D/Daum/presence.ts [hunk 10]
```diff
@@ -0,0 +1,447 @@
...
+          if (video){
+            const timestamps = presence.getTimestamps(video.currentTime, video.duration);
+
+            presenceData.details = "Watching video:";
+            presenceData.state = video.title ?? document.querySelector("h3.tit_view")?.textContent;
+
+            presenceData.smallImageKey = video.paused ? "pause" : "play";
+            presenceData.smallImageText = video.paused ? "Paused" : "Playing";
+
+            presenceData.startTimestamp = timestamps[0];
```
Comment (ririxidev) [inline websites/D/Daum/presence.ts:133]: You don't need to specify startTimestamp when you have endTimestamp...

File: websites/D/Daum/presence.ts [hunk 11]
```diff
@@ -0,0 +1,447 @@
...
+  cafe?: typeof cafe
+}) => { 
+  if (data.video) video = data.video;
+  if (data.cafe) cafe = data.cafe;
+});
+
+presence.on("UpdateData", async () => {
+    const presenceData: PresenceData = {
+      largeImageKey: "daum",
+      details: "Browsing...",
```
Comment (ririxidev) [inline websites/D/Daum/presence.ts:94]: is it really necessary when you did support for every subpage?

**Review by ririxidev — CHANGES_REQUESTED** [review]
Heya @SlowLife1661!
You should check out thing(s) that I requested to change in my review. Please look at them!

Regards,
**ririxi**. PreMiD Reviewer Team.
Contact: mail@ririxi.dev, ririxi#2721.

File: websites/D/Daum/iframe.ts (hunk 4 above)
Comment (SlowLife1661) [reply websites/D/Daum/iframe.ts:18]: I mean it won't give me any data if I don't use iframe

File: websites/D/Daum/presence.ts (hunk 11 above)
Comment (SlowLife1661) [reply websites/D/Daum/presence.ts:94]: idk but it just does looks cooler *xd*

File: websites/D/Daum/iframe.ts (hunk 4 above)
Comment (ririxidev) [reply websites/D/Daum/iframe.ts:18]: strange.

File: websites/D/Daum/presence.ts (hunk 11 above)
Comment (ririxidev) [reply websites/D/Daum/presence.ts:94]: but ~~it doesn't~~

Comment (SlowLife1661) [reply websites/D/Daum/presence.ts]: meanie

**Review by ririxidev — APPROVED** [review]
ok

**Review by Bas950 — APPROVED** [review]
LGTM

**Pull request merged**